src/classes/StatusScheduler.cls

```diff
@@ -0,0 +1,57 @@
+global class StatusScheduler implements  Schedulable {
+List<Assignment__c> toUpdate = new List<Assignment__c>();
+List<Contact> toUpdateContact = new List<Contact>();
+    
+    global void  execute(SchedulableContext sc){
```

Comment:
It is actually better if you move this code into a class that implements Database.Batchable because that way your class can process more records at a time. After that is done you can use this class to setup a schedule in which the Batchable one will be called.